crates/iceberg/src/spec/partition.rs

```diff
@@ -44,22 +44,27 @@ pub struct PartitionField {
     pub transform: Transform,
 }
 
+impl PartitionField {
+    /// To unbound partition field
+    pub fn into_unbound(self) -> UnboundPartitionField {
+        self.into()
+    }
+}
+
 ///  Partition spec that defines how to produce a tuple of partition values from a record.
-#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone, Default, Builder)]
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone, Default)]
```

Comment:
We could only add schema_id as `Option<i32>` as `TableMetadata` does not store the schema_id as part of its PartitionSpecs - thus we won't be able to deserialize this information. I am not sure if it's worth adding it - it would only be relevant while building in memory.

I fully agree with not having the attributes public (see also my initial statement point 6 at the very top). I addressed it in the latest commit.